tests/test_scheduler_pickle.py

```diff
@@ -32,6 +32,21 @@ def test_bare_scheduler(self):
         self.assertIsInstance(scheduler, CoreScheduler)
         self.assertIsInstance(conditions, Conditions)
 
+    def test_extra_scheduler(self):
+        # Test if we can read a pickle with extra stuff in it.
+        scheduler, conditions = read_scheduler()
+        extra_content = "I am a fish."
+        with TemporaryDirectory() as data_dir:
+            content = (scheduler, conditions, extra_content)
+            sample_path = os.path.join(data_dir, "sample_scheduler.pickle")
+            with open(sample_path, "wb") as file_io:
+                pickle.dump(content, file_io)
+
+            scheduler, conditions = read_scheduler(sample_path)
+
+        self.assertIsInstance(scheduler, CoreScheduler)
+        self.assertIsInstance(conditions, Conditions)
+
```

Comment:
Maybe want to also check that `extra_content` is able to be restored just for completeness.

Reply:
Okay, I've added an option just to get everything (but I don't use this anywhere).